src/resolvers/line_items.ts

```diff
@@ -72,15 +89,57 @@ const addLineItem: MutationResolvers['addLineItem'] = async (
   })
 }
 
+// - update line item
+
+const updateLineItemInput = Type.Object(
+  {
+    id: ID,
+    input: Type.Object(
+      {
+        status: Type.Optional(Type.Enum(LineItemStatus)),
+        proposedReceivingGroupId: Type.Optional(ID),
+        acceptedReceivingGroupId: Type.Optional(ID),
```

Comment:
Issue: @coderbyheart there's a problem here because the frontend will send `null` for some optional fields, but the `Type.Optional` validations only allow either `undefined` or a specific type.

Suggestions:
- we could either format the data differently on the frontend (convert all `null` values to `undefined` or delete `null` values from the payload), or
- we could change the validation. I'm not super familiar with Typebox but I saw a `Type.Null()` option.

What do you think is best? I have some time this week to work on the frontend code if that helps.

Reply:
I am not a fan of sending blank values over the wire, so if a value is not present, it should be omitted from the JSON sent by the fronted (set to `undefined`).

We can also add the contract to the API that the frontend can sent a value as `null | undefined` and we can apply a `nullToUndefined` filter before passing the input to the validation. 

I'd prefer the first.

Reply:
Another reason for not sending `NULL` generally to signal absence of input is that it is used in some cases to delete values server side: 

https://github.com/distributeaid/shipment-tracker/blob/abd84308879f8eee7d55bb26303b44b176672e97/src/tests/groups_api.test.ts#L194-L205

Reply:
I agree with you. I've pushed a quick fix to this PR, and I'll work on a `nullToUndefined` filter in a future PR.